modules/pam_keyinit/pam_keyinit.c

```diff
@@ -162,11 +162,7 @@ static void kill_keyrings(pam_handle_t *pamh)
 	}
 }
 
-/*
- * open a PAM session by making sure there's a session keyring
- */
-int pam_sm_open_session(pam_handle_t *pamh, int flags UNUSED,
-			int argc, const char **argv)
+static int do_open_session(pam_handle_t *pamh, int argc, const char **argv, int ERROR)
```

Comment:
Name the function differently - I'd suggest do_keyinit or something similar. Also please note that the session_counter should not be incremented when calling from pam_sm_setcred.